src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -15,8 +15,9 @@ namespace NuGet.Packaging.Signing
 {
     public static class SignedPackageArchiveIOUtility
     {
-        private const int _bufferSize = 4096;
-
+        [ThreadStatic]
+        private static byte[] _sharedBuffer;
```

Comment:
I could not see any existing pooling of objects or arrays via ObjectPool<T> or something similar, so I introduced a poor-man's version.